src/utils/hwWallet.js

```diff
@@ -10,37 +10,57 @@ import { getTransactionBytes, calculateTxId, getBufferToHex } from './rawTransac
 import { PLATFORM_TYPES, getPlatformType } from './platform';
 import store from '../store';
 import loginTypes from '../constants/loginTypes';
+import { accountLoggedOut } from '../actions/account';
+import { dialogDisplayed, dialogHidden } from '../actions/dialog';
+import Alert from '../components/dialog/alert';
 
 const util = require('util');
 
 const { ipc } = window;
 
+const handleConnect = () => {
+  store.dispatch({
+    type: actionTypes.settingsUpdated,
+    data: { isHarwareWalletConnected: true },
+  });
+  store.dispatch(errorToastDisplayed({ label: HW_MSG.LEDGER_CONNECTED }));
+};
+
+const handleDisconnect = (hwName) => {
+  const state = store.getState();
+  const { account } = state;
+  if (account.address) {
+    store.dispatch( // eslint-disable-line
+      dialogDisplayed({
+        childComponent: Alert,
+        childComponentProps: {
+          title: 'You are disconnected',
+          text: `There is no connection to the ${hwName}. Please check the cables if it happened by accident.`,
+          closeDialog: () => {
+            store.dispatch(dialogHidden());
+            location.reload(); // eslint-disable-line
+          },
+        },
+      }));
+    store.dispatch(accountLoggedOut());
+  }
+  store.dispatch({
+    type: actionTypes.settingsUpdated,
+    data: { isHarwareWalletConnected: false },
+  });
+};
+
 if (ipc) { // On browser-mode is undefined
-  // ipc.on('ledgerConnected', () => {
-  //   store.dispatch(infoToastDisplayed({ label: HW_MSG.LEDGER_CONNECTED }));
-  // });
+  ipc.on('ledgerConnected', handleConnect);
+  ipc.on('ledgerDisconnected', () => handleDisconnect('Ledger Nano S'));
 
-  // ipc.on('ledgerDisconnected', () => {
-  //   console.log('ledgerDisconnected');
-  //   store.dispatch(errorToastDisplayed({ label: HW_MSG.LEDGER_DISCONNECTED }));
-  // });
+  ipc.on('trezorConnected', handleConnect);
+  ipc.on('trezorDisconnected', () => handleDisconnect('Trezor Model T'));
```

Comment:
IMO here is an unnecessary duplication of having separate events for ledger and trezor. 
It can already on the electron side dispatch the same event with model, e.g.:
change
https://github.com/LiskHQ/lisk-hub/blob/29871a8a23f32391ac9f06adb4a9ab7492feaebe/app/src/ledger.js#L95
to 
```
win.send({ event: 'hwWalletDisconnected', value: {model: 'Trezor Model T' });
```

And then here:
```suggestion
  ipc.on('trezorDisconnected', (event, { model }) => handleDisconnect(model));
```

Relevant docs for Electron ipc: https://electronjs.org/docs/api/ipc-main#ipcmainonchannel-listener

Reply:
thx Vit I made the correction you are right about it